### PR TITLE
non-aggregation query support

### PR DESCRIPTION
**Pull request by voyager-dw**
will add more test cases for go part

### CODE REVIEW

File: query/aql_nonaggr_batchexecutor.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+// Run is the function to run the whole process for a batch
+func (e *NonAggrBatchExecutorImpl) Run(isLastBatch bool) {
+	e.isLastBatch = isLastBatch
+	start := utils.Now()
+	// initialize index vector.
+	initIndexVector(e.qc.OOPK.currentBatch.indexVectorD.getPointer(), 0, e.qc.OOPK.currentBatch.size, e.stream, e.qc.Device)
+
+	e.qc.reportTimingForCurrentBatch(e.stream, &start, initIndexVectorTiming)
+
+	e.filter()
```
Comment (shz117) [inline query/aql_nonaggr_batchexecutor.go]: missing implementation?

Comment (shz117) [reply query/aql_nonaggr_batchexecutor.go]: nvm it will get from embedded batch executor

File: query/iterator_unittest.cu [hunk 2]
```diff
@@ -663,4 +663,24 @@ TEST(GeoPredicateIteratorTest, CheckIterator) {
...
+  uint32_t counts[10] = {1,3,5,8,9,10,15,16,18,20};
+  uint32_t indexes[3] = {2,5,3};
+
+  IndexCountIterator it(counts, indexes);
+  EXPECT_EQ(it[0], 3);
+  EXPECT_EQ(it[1], 5);
+  EXPECT_EQ(it[2], 1);
+}
+
+int expand(uint32_t *baseCounts,
```
Comment (shz117) [inline query/iterator_unittest.cu]: maybe remove this call if it's not tested in this func

File: query/time_series_aggregate.h [hunk 3]
```diff
@@ -20,6 +20,21 @@
 #include <stdint.h>
 #include "../cgoutils/utils.h"
 
+
+// some macro used for host or device compilation
+#ifdef RUN_ON_DEVICE
```
Comment (shz117) [inline query/time_series_aggregate.h]: this is neat 👍

File: query/aql_batchexecutor.go [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+	// project of measure/select columns
+	project()
+	// reduce to sort and aggregate result
+	reduce()
+	// main method to call for execution
+	Run(isLastBatch bool)
+}
+
+// DummyBatchExecutorImpl is a dummy executor which do nothing
+type DummyBatchExecutorImpl struct {
```
[the remark this answers is not in the document]
Comment (shz117) [reply query/aql_batchexecutor.go]: it's used for no-op placeholder at begining of pipeline. see where `NewDummyBatchExecutor` is called

**Comment by voyager-dw** [comment]
support non-aggregation query

**Pull request merged**